### PR TITLE
[381728] prevent NPE in AdvancedTemplatesPreferencePage

### PR DESCRIPTION
**Pull request by cdietrich**
[381728] prevent NPE in AdvancedTemplatesPreferencePage

Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=381728

**Comment by cdietrich** [comment]
please review

File: org.eclipse.xtext.ui.codetemplates.ui/src/org/eclipse/xtext/ui/codetemplates/ui/preferences/AdvancedTemplatesPreferencePage.java [hunk 1]
```diff
@@ -66,14 +66,17 @@ protected void updateViewerInput() {
...
+				String prefix = 
+						"templates for " + grammarAccess.getGrammar().getName() +
+						"'" + name + "'" + " for " + getContextTypeForGrammar(contextType) + ">>";
+				String editablePart = template.getPattern();
+				String suffix = "";
+				partialEditor.updateModel(prefix, editablePart, suffix);
+			} else {
+				partialEditor.updateModel("", template.getPattern(), "");
+			}
 		} else {
```
Comment (svenefftinge) [inline org.eclipse.xtext.ui.codetemplates.ui/src/org/eclipse/xtext/ui/codetemplates/ui/preferences/AdvancedTemplatesPreferencePage.java]: empty else?

Comment (cdietrich) [inline org.eclipse.xtext.ui.codetemplates.ui/src/org/eclipse/xtext/ui/codetemplates/ui/preferences/AdvancedTemplatesPreferencePage.java]: well i wanted the user to give a chance to "repair" his/her template

Comment (cdietrich) [inline org.eclipse.xtext.ui.codetemplates.ui/src/org/eclipse/xtext/ui/codetemplates/ui/preferences/AdvancedTemplatesPreferencePage.java]: hmmm ill have a look

**Comment by cdietrich** [comment]
@svenefftinge i updated the pullreq

**Pull request merged**